manifests/copr.pp

```diff
@@ -0,0 +1,71 @@
+#
+# @summary This definition manages Copr (Cool Other Package Repo) repositories.
+#
+# @param copr_repo name of repository, defaults to title
+# @param ensure specifies if repo should be enabled, disabled or removed
+#
+# @example add and enable COPR restic repository
+#   yum::copr { 'copart/restic':
+#     ensure  => 'enabled',
+#   }
+#
+define yum::copr (
+  String                                 $copr_repo = $title,
+  Enum['enabled', 'disabled', 'removed'] $ensure    = 'enabled',
+) {
+  $prereq_plugin = $facts['package_provider'] ? {
+    'dnf'   => 'dnf-plugins-core',
+    default => 'yum-plugin-copr',
+  }
+  stdlib::ensure_packages([$prereq_plugin])
```

Comment:
please introduce a new param like $manage_prereq_plugin and wrap an if condition around thr function.

Reply:
That makes a lot of sense, done and force-pushed.